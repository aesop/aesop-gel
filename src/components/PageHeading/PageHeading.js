import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Heading from '~/components/Heading';
import { getHeadingLevel, getSubHeadingLevel } from './PageHeading.utils';
import styles from './PageHeading.module.css';

const PageHeading = ({
  childrenClassNames,
  className,
  eyebrow,
  id,
  subHeading,
  heading,
}) => {
  const classSet = cx(styles.base, className);
  const eyebrowClassSet = cx(styles.eyebrow, childrenClassNames.eyebrow);
  const headingClassSet = cx(styles.heading, childrenClassNames.heading);
  const subHeadingClassSet = cx(
    styles.subHeading,
    childrenClassNames.subHeading,
  );

  return (
    <header className={classSet} id={id}>
      {eyebrow && (
        <Heading className={eyebrowClassSet} level="1" size="xSmall">
          {eyebrow}
        </Heading>
      )}
      {heading && (
        <Heading
          className={headingClassSet}
          level={getHeadingLevel(eyebrow)}
          size="large"
        >
          {heading}
        </Heading>
      )}
      {subHeading && (
        <Heading
          className={subHeadingClassSet}
          level={getSubHeadingLevel(eyebrow, heading)}
          size="xSmall"
        >
          {subHeading}
        </Heading>
      )}
    </header>
  );
};

PageHeading.propTypes = {
  childrenClassNames: PropTypes.shape({
    eyebrow: PropTypes.string,
    heading: PropTypes.string,
    subHeading: PropTypes.string,
  }),
  className: PropTypes.string,
  eyebrow: PropTypes.string,
  heading: PropTypes.string,
  id: PropTypes.string,
  subHeading: PropTypes.string,
};

PageHeading.defaultProps = {
  childrenClassNames: {
    eyebrow: undefined,
    heading: undefined,
    subHeading: undefined,
  },
  className: undefined,
  eyebrow: undefined,
  heading: undefined,
  id: undefined,
  subHeading: undefined,
};

export default PageHeading;
