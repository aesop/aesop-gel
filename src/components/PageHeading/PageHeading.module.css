@import 'styles/variables/index';

.base {
  position: relative;
  margin-top: rem(22px);
  margin-bottom: rem(22px);
}

.heading {
  margin: 0 0 20px;
}

.eyebrow {
  color: var(--color-body-copy);
}

.subHeading {
  color: var(--color-body-copy);
}
