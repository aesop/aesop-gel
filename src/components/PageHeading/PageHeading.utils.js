export const getHeadingLevel = eyebrow => (eyebrow ? '3' : '2');

export const getSubHeadingLevel = (eyebrow, heading) => {
  if (eyebrow) {
    return heading ? '4' : '3';
  }

  return heading ? '3' : '2';
};
