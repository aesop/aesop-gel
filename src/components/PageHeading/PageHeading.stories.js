import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import '~.storybook/storybook.module.css';
import PageHeading from './PageHeading';

storiesOf('Components/PageHeading', module).add('Base component', () => (
  <PageHeading
    eyebrow={text(
      'eyebrow',
      'Eyebrow: Offers fortification of the highest order against free radicals.',
    )}
    heading={text(
      'heading',
      'Heading: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream.',
    )}
    id="Anti-Oxidant"
    subHeading={text(
      'subHeading',
      'SubHeading: Parsley Seed Cream. Intensely soothing, nourishing and hydrating, this elegant formulation contains a potent blend of botanicals that offer fortification of the highest order against free radicals.',
    )}
  />
));
