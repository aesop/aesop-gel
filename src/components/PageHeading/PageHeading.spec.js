import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import PageHeading from './PageHeading';

configure({ adapter: new Adapter() });

describe('<PageHeading />', () => {
  it('should be defined', () => {
    expect(PageHeading).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer
      .create(
        <PageHeading
          heading="Title: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream."
          id="Anti-Oxidant"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with subHeading variation correctly', () => {
    const tree = renderer
      .create(
        <PageHeading
          heading="Heading: Fortification of the highest order."
          id="Parsley"
          subHeading="Subcopy: Parsley Seed Anti-Oxidant Facial Hydrating Cream 60mL"
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with subHeading and eyebrow variation correctly', () => {
    const tree = renderer
      .create(
        <PageHeading
          eyebrow="Eyebrow: Offers fortification of the highest order against free radicals."
          heading="Heading: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream."
          id="Anti-Oxidant"
          subHeading="SubHeading: Parsley Seed Cream. Intensely soothing, nourishing and hydrating, this elegant formulation contains a potent blend of botanicals that offer fortification of the highest order against free radicals."
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
