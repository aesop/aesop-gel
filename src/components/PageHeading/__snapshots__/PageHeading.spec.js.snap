// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeading /> renders base component correctly 1`] = `
<header
  className="base"
  id="Anti-Oxidant"
>
  <h2
    className="base large heading"
  >
    Title: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream.
  </h2>
</header>
`;

exports[`<PageHeading /> renders with subHeading and eyebrow variation correctly 1`] = `
<header
  className="base"
  id="Anti-Oxidant"
>
  <h1
    className="base xSmall eyebrow"
  >
    Eyebrow: Offers fortification of the highest order against free radicals.
  </h1>
  <h3
    className="base large heading"
  >
    Heading: Fortification of the highest order. Parsley Seed Anti-Oxidant Facial Hydrating Cream.
  </h3>
  <h4
    className="base xSmall subHeading"
  >
    SubHeading: Parsley Seed Cream. Intensely soothing, nourishing and hydrating, this elegant formulation contains a potent blend of botanicals that offer fortification of the highest order against free radicals.
  </h4>
</header>
`;

exports[`<PageHeading /> renders with subHeading variation correctly 1`] = `
<header
  className="base"
  id="Parsley"
>
  <h2
    className="base large heading"
  >
    Heading: Fortification of the highest order.
  </h2>
  <h3
    className="base xSmall subHeading"
  >
    Subcopy: Parsley Seed Anti-Oxidant Facial Hydrating Cream 60mL
  </h3>
</header>
`;
